src/modules/FioAddress/action.js

```diff
@@ -25,17 +25,24 @@ export const refreshAllFioAddresses = () => async (dispatch: Dispatch, getState:
   const wallets: EdgeCurrencyWallet[] = getFioWallets(getState())
   let fioAddresses = []
 
-  if (wallets != null) {
+  if (wallets != null && wallets.length) {
     for (const wallet: EdgeCurrencyWallet of wallets) {
       const walletFioAddresses = await wallet.otherMethods.getFioAddresses()
       fioAddresses = [...fioAddresses, ...walletFioAddresses]
     }
-  }
 
-  dispatch({
-    type: 'FIO/SET_FIO_ADDRESSES',
-    data: { fioAddresses }
-  })
+    dispatch({
+      type: 'FIO/SET_FIO_ADDRESSES',
+      data: { fioAddresses }
+    })
+  } else {
+    window.requestAnimationFrame(() => {
```

Comment:
Why not just use `window.requestAnimationFrame` around the dispatch on line 34 therefore always using it ?

Reply:
yeah, we can use it always, just weird thing and not a beautiful solution as for me
will update it now